examples/asr/transcribe_speech.py

```diff
@@ -255,6 +257,14 @@ def main(cfg: TranscriptionConfig) -> Union[TranscriptionConfig, List[Hypothesis
     asr_model.set_trainer(trainer)
     asr_model = asr_model.eval()
 
+    if cfg.compute_dtype != "float32" and cfg.amp:
+        raise ValueError("amp=true is mutually exclusive with a compute_dtype other than float32")
+
+    amp_dtype = torch.float16 if cfg.amp_dtype == "float16" else torch.bfloat16
```

Comment:
Lets change default to bf16 in another PR later. Fyi @nithinraok

Reply:
Make sense, @galv could you please also add this to the PR. Thanks!

Reply:
the default is float32 right now. bfloat16 can be made the default later.

Reply:
To clarify, I don't want to change the default value right now. It would cause transcribe_speech.py to throw an error for all users who also turn on the amp=true.